columnphysics/icepack_flux.F90

```diff
@@ -220,9 +220,8 @@ subroutine merge_fluxes (aicen,                &
       if (snwgrain) then
          meltsliq  = meltsliq  + meltsliqn * aicen
       endif
-      if (present(dsnow)) then
+      if (present(dsnow)) &
```

Comment:
This probably should check present of dsnow and dsnown like others above.  Also fswthru_* maybe needs the same.  We might also want to improve robustness.  If dsnow is passed but dsnown isn't (or vv), should this throw an error?  Again, same question for all other optional diagnostics.  Those checks could be at the start of the subroutine.